sympy/physics/mechanics/kane.py

```diff
@@ -290,33 +292,35 @@ def _initialize_constraint_matrices(self, config, vel, acc, linear_solver='LU'):
             raise ValueError('There must be an equal number of dependent '
                              'speeds and acceleration constraints.')
         if vel:
-            u_zero = dict.fromkeys(self.u, 0)
-            udot_zero = dict.fromkeys(self._udot, 0)
-
-            # When calling kanes_equations, another class instance will be
-            # created if auxiliary u's are present. In this case, the
-            # computation of kinetic differential equation matrices will be
-            # skipped as this was computed during the original KanesMethod
-            # object, and the qd_u_map will not be available.
-            if self._qdot_u_map is not None:
-                vel = msubs(vel, self._qdot_u_map)
-
-            self._f_nh = msubs(vel, u_zero)
-            self._k_nh = (vel - self._f_nh).jacobian(self.u)
-            # If no acceleration constraints given, calculate them.
-            if not acc:
-                _f_dnh = (self._k_nh.diff(dynamicsymbols._t) * self.u +
-                          self._f_nh.diff(dynamicsymbols._t))
+            try:
```

Comment:
Good to see that you've figured out how to use the try except statement. However, we don't really require it here as the error is already clear enough. You can vastly simplify all changes to:
```python
    def _initialize_constraint_matrices(self, config, vel, acc, linear_solver='LU'):
        """Initializes constraint matrices."""
        ...
        if vel:
            # When calling kanes_equations, another class instance will be
            # created if auxiliary u's are present. In this case, the
            # computation of kinetic differential equation matrices will be
            # skipped as this was computed during the original KanesMethod
            # object, and the qd_u_map will not be available.
            if self._qdot_u_map is not None:
                vel = msubs(vel, self._qdot_u_map)

            self._k_nh, self._f_nh = linear_eq_to_matrix(vel, self.u[:])
            # If no acceleration constraints given, calculate them.
            if not acc:
                ...
            else:
                if self._qdot_u_map is not None:
                    acc = msubs(acc, self._qdot_u_map)
                self._k_dnh, self._f_dnh = linear_eq_to_matrix(acc, self._udot[:])

            ...
        else:
            ...
```

Reply:
I am getting all sorts of errors! :-(
You think, they are related to my try.....except?

Reply:
Hmmm, we are apparently defining things as `self._k_nh * self.u + self._f_nh == vel`, while `linear_eq_to_matrix` uses the form `Ax - b == eqs`.

P.S. I just updated my above code suggestion.

Reply:
> Hmmm, we are apparently defining things as `self._k_nh * self.u + self._f_nh == vel`, while `linear_eq_to_matrix` uses the form `Ax - b == eqs` (if I'm not mistaken).
> 
> P.S. I've just updated my above code suggestion.

I have tried this, but still so many mistakes. :-((
But it is so much faster it has to work!

Reply:
I'm having no problems. Here is a [patch](https://github.com/user-attachments/files/17210440/peter.patch) (the patch is w.r.t. master). You can either just view the contents of the file or checkout and apply it to master.